src/Support.php

```diff
@@ -0,0 +1,45 @@
+<?php
+namespace Krokedil\Support;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Class Support.
+ */
+class Support {
+
+	/**
+	 * Logger instance.
+	 *
+	 * @var Logger
+	 */
+	private $logger = null;
+
+	/**
+	 * System report class.
+	 *
+	 * @var SystemReport
+	 */
+	private $system_report = null;
+
+	/**
+	 * Support constructor.
+	 *
+	 * @param string $id Gateway ID.
+	 * @param string $name Gateway name (or title).
+	 */
+	public function __construct( $id, $name ) {
+		$this->logger        = new Logger( $id );
+		$this->system_report = new SystemReport( $id, $name );
+	}
```

Comment:
Remove the constructor and rather add methods to create instances of the logger and system report class manually by the person implementing the package. This makes it so that people implementing the package are not forced to use all the features the package provides if its not required or wanted. It will also help in the long run when we add more features, so we don't have to keep extending the class constructor here with new parameters that those classes will need for the constructor, rather we just add a new method for creating the instance.